src/test/java/de/tum/in/www1/artemis/iris/PyrisLectureIngestionTest.java

```diff
@@ -0,0 +1,104 @@
+        }
+        userUtilService.addUsers(TEST_PREFIX, 1, 1, 0, 1);
+        List<Course> courses = courseUtilService.createCoursesWithExercisesAndLectures(TEST_PREFIX, true, true, 1);
+        Course course1 = this.courseRepository.findByIdWithExercisesAndExerciseDetailsAndLecturesElseThrow(courses.getFirst().getId());
+        this.lecture1 = course1.getLectures().stream().findFirst().orElseThrow();
+        this.lecture1.setTitle("Lecture " + lecture1.getId()); // needed for search by title
+        this.lecture1 = lectureRepository.save(this.lecture1);
+        // Add users that are not in the course
+        userUtilService.createAndSaveUser(TEST_PREFIX + "student42");
+        userUtilService.createAndSaveUser(TEST_PREFIX + "tutor42");
+        userUtilService.createAndSaveUser(TEST_PREFIX + "instructor42");
+
+        int numberOfSlides = 2;
+        AttachmentUnit attachmentUnitWithSlides = lectureUtilService.createAttachmentUnitWithSlidesAndFile(numberOfSlides);
+        lecture1 = lectureUtilService.addLectureUnitsToLecture(lecture1, List.of(attachmentUnitWithSlides));
+        this.lecture1 = lectureRepository.findByIdWithLectureUnitsAndAttachmentsElseThrow(lecture1.getId());
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void testPyrisLectureIngestionWebhookJob_isEmpty_When_subsettingsDisabled() throws Exception {
+        Course course2 = courseUtilService.addEmptyCourse();
+        courseUtilService.enableMessagingForCourse(course2);
+        request.postWithResponseBody("/api/lectures/import/" + lecture1.getId() + "?courseId=" + course2.getId(), null, Lecture.class, HttpStatus.CREATED);
+        if (lecture1.getLectureUnits().getFirst() instanceof AttachmentUnit attachmentUnit) {
+            pyrisWebhookService.executeLectureIngestionPipeline(true, List.of(attachmentUnit));
+        }
+        Set<String> keySet = jobService.getAllJobs().keySet();
+        assertThat(keySet).isEmpty();
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void testExecuteLectureIngestionPipeline_isNotEmpty_When_subsettingsEnabled() throws Exception {
+        Course course2 = courseUtilService.addEmptyCourse();
+        courseUtilService.enableMessagingForCourse(course2);
+        activateIrisFor(course2);
+        irisRequestMockProvider.mockIngestionWebhookRunResponse(dto -> {
+            assertThat(dto.settings().authenticationToken()).isNotNull();
+        });
+        request.postWithResponseBody("/api/lectures/import/" + lecture1.getId() + "?courseId=" + course2.getId(), null, Lecture.class, HttpStatus.CREATED);
+        if (lecture1.getLectureUnits().getFirst() instanceof AttachmentUnit attachmentUnit) {
+            pyrisWebhookService.executeLectureIngestionPipeline(true, List.of(attachmentUnit));
+        }
+        Set<String> keySet = jobService.getAllJobs().keySet();
+        assertThat(keySet).isNotEmpty();
```

Comment:
This would only check if the job was started. Also it is like the other one not safe.

I'd also like to see some more tests that check:
- Successful status update was sent
- Error status update was sent
- Possibly some more edge cases (pyris down, job token doesn't exist (any more))